### PR TITLE
Improve the Library example (with persistent)

### PR DESCRIPTION
**Pull request by VladimirLogachev**
Hi!

My goal is to have a comprehensive example to use with different clients (such as with `elm-graphql` and other languages). So I changed this:

- Add `newBook` mutation
- Insert demo data to be instantly available when server starts (same demo data as in `library-graphql` executable) 
- Add uniqueness constraint for a tuple of a book title and book author id

### CODE REVIEW

File: examples/library/src/Server.hs
Comment (kutyel) [inline examples/library/src/Server.hs:72]:
```suggestion
  (`catchError` (const $ pure Nothing)) . runDb conn . fmap sequence_ $ do
```

File: examples/library/src/Server.hs
Comment (kutyel) [inline examples/library/src/Server.hs:74]:
```suggestion
    traverse (insertUnique . ($ authorId)) books
```

**Review by kutyel — CHANGES_REQUESTED** [review]
Looks great! 👏  I only added a few minor stylistic changes to match the rest of the mu-haskell codebase 😉

**Review by kutyel — DISMISSED** [review]
🚀  @serras what do you think? :)

File: examples/library/src/Server.hs
Comment (serras) [inline examples/library/src/Server.hs:69]: Would it be possible to re-write this function in a simpler way? Maybe using pattern matching on `authorResult`, as it was at some point?

Our goal with our examples is to showcase the library, but the readers may not be Haskell experts. Thus we try to keep the level of the language as low as possible.

File: examples/library/src/Server.hs
Comment (serras) [inline examples/library/src/Server.hs:72]: Similar here, we could write a much easier version by eta-expanding.

```haskell
   insertUnique . ($ authorId)
= \kBook -> insertUnique (kBook $ authorId)
= \kBook -> insertUnique (kBook authorId)
```

**Review by serras — CHANGES_REQUESTED** [review]

File: examples/library/src/Server.hs
Comment (VladimirLogachev) [reply examples/library/src/Server.hs:69]: Sure!

File: examples/library/src/Server.hs
Comment (VladimirLogachev) [reply examples/library/src/Server.hs]: Changed it, thank you.

**Review by kutyel — APPROVED** [review]

**Review by serras — APPROVED** [review]

**Comment by serras** [comment]
Thanks for your help! I'll merge once CI passes.

**Comment by VladimirLogachev** [comment]
Is there an issue with CI? Maybe it is not going to run on forks?

**Comment by VladimirLogachev** [comment]
If it's true that CI pipeline isn't going to run automatically on forks, and also can not be run manually, then maybe we may have a special branch for that? Call it `incoming`, `ci`, `integration` (or like that) for one specific reason: run CI pipeline before merging into master. Then such a branch could become a target branch for this PR.

**Comment by kutyel** [comment]
Or we can just merge it and see what happens 😂  thanks for your work!

**Pull request merged**
